### DISCUSSION TITLE
Server Error when trying to add a new component

### DISCUSSION THREAD

**Discussion by Tefel** [WeblateOrg/weblate#6559] [Q&A] [answered]
Hello fellow devs! 
I installed my own Weblate following the guide: https://docs.weblate.org/en/latest/admin/install/venv-debian.html
I was able to create an account and the project. There is a Server Error when trying to add a new component:

![image](https://user-images.githubusercontent.com/1677244/133641927-a9416f95-ea71-4a16-8139-acb9835d6c20.png)

Plese check my running **Weblate** here (if you need credentials let me know)
https://astrocolony1.vps.webdock.io/create/component/

**There are some warnings when starting the server**
> ?: (weblate.W025.ass) Failure in loading handler for ass file format: aeidon or gaupol package required for Subtitle support
> HINT: https://docs.weblate.org/en/weblate-4.8/admin/install.html#optional-deps
> ?: (weblate.W025.fluent) Failure in loading handler for fluent file format: No module named 'fluent'
> HINT: https://docs.weblate.org/en/weblate-4.8/admin/install.html#optional-deps
> ?: (weblate.W025.ini) Failure in loading handler for ini file format: Missing iniparse library.
> HINT: https://docs.weblate.org/en/weblate-4.8/admin/install.html#optional-deps
> ?: (weblate.W025.islu) Failure in loading handler for islu file format: Missing iniparse library.
> HINT: https://docs.weblate.org/en/weblate-4.8/admin/install.html#optional-deps
> ?: (weblate.W025.laravel) Failure in loading handler for laravel file format: No module named 'phply'
> HINT: https://docs.weblate.org/en/weblate-4.8/admin/install.html#optional-deps
> ?: (weblate.W025.php) Failure in loading handler for php file format: No module named 'phply'
> HINT: https://docs.weblate.org/en/weblate-4.8/admin/install.html#optional-deps
> ?: (weblate.W025.srt) Failure in loading handler for srt file format: aeidon or gaupol package required for Subtitle support
> HINT: https://docs.weblate.org/en/weblate-4.8/admin/install.html#optional-deps
> ?: (weblate.W025.ssa) Failure in loading handler for ssa file format: aeidon or gaupol package required for Subtitle support
> HINT: https://docs.weblate.org/en/weblate-4.8/admin/install.html#optional-deps
> ?: (weblate.W025.sub) Failure in loading handler for sub file format: aeidon or gaupol package required for Subtitle support
> HINT: https://docs.weblate.org/en/weblate-4.8/admin/install.html#optional-deps
> ?: (weblate.W033.Gerrit) Failure in loading VCS module for Gerrit: git: 'review' is not a git command. See 'git --help'.
> (1)
> HINT: https://docs.weblate.org/en/weblate-4.8/vcs.html
> ?: (weblate.W033.Mercurial) Failure in loading VCS module for Mercurial: [Errno 2] No such file or directory: 'hg'
> HINT: https://docs.weblate.org/en/weblate-4.8/vcs.html
> ?: (weblate.W033.Subversion) Failure in loading VCS module for Subversion: git: 'svn' is not a git command. See 'git --help'.
> 
> System check identified 12 issues (1 silenced).
> September 12, 2021 - 08:10:04 AM
> Django version 3.2.7, using settings 'weblate.settings'
> Starting development server at http://127.0.0.1:8080/

**About my configuration**
Weblate 4.8 Django 3.2.7 siphashc 2.1 translate-toolkit 3.4.0 lxml 4.6.3 Pillow 8.3.2 bleach 4.1.0 python-dateutil 2.8.2 social-auth-core 4.1.0 social-auth-app-django 5.0.0 django-crispy-forms 1.12.0 oauthlib 3.1.1 django-compressor 2.4.1 djangorestframework 3.12.4 django-filter 2.4.0 django-appconf 1.0.4 user-agents 2.2.0 filelock 3.0.12 setuptools 44.0.0 jellyfish 0.8.2 openpyxl 3.0.7 celery 5.1.2 kombu 5.1.0 translation-finder 2.10 weblate-language-data 2021.5 html2text 2020.1.16 pycairo 1.20.1 pygobject 3.40.1 diff-match-patch 20200713 requests 2.26.0 django-redis 5.0.0 hiredis 2.0.0 sentry_sdk 1.3.1 Cython 0.29.24 misaka 2.1.1 GitPython 3.1.18 borgbackup 1.1.17 pyparsing 2.4.7 pyahocorasick 1.4.2 python-redis-lock 3.7.0 Python 3.8.10 Git 2.25.1 psycopg2-binary 2.9.1 ruamel.yaml 0.17.16 mysqlclient 2.0.3 Redis server 5.0.7 MariaDB sever 5.5.5

Hope you can help me find out the problem! I cannot find it by myself. I have no idea what can be wrong with the configuration or a server. Please help! I hope it is resolvable. I am willing to support the project with donation if needed!

**Comment by nijel** [comment]
Please check the logs and/or e-mail for more detailed error report.

**Reply by Tefel** [reply]
Hey, great work, thanks for an answer! Here is the log when I try to get into "create component" tab:
> System check identified 12 issues (1 silenced).
> September 19, 2021 - 08:27:20 PM
> Django version 3.2.7, using settings 'weblate.settings'
> Starting development server at http://127.0.0.1:8080/
> Quit the server with CONTROL-C.
> [19/Sep/2021 20:27:21] "GET / HTTP/1.0" 200 29015
> [19/Sep/2021 20:27:28] "GET /create/component/ HTTP/1.0" 500 12023

**Reply by nijel** [reply]
Hmm, the error exception should be printed to the log, did you modify logging configuration?

Alternative ways to get full exception: https://docs.weblate.org/en/latest/admin/install.html#properly-configure-admins or https://docs.weblate.org/en/latest/admin/install.html#collecting-error-reports

Also see https://docs.weblate.org/en/latest/contributing/debugging.html

**Reply by Tefel** [reply]
Hey thanks, I had debug mode disabled.
![image](https://user-images.githubusercontent.com/1677244/134394480-ef84d28a-911d-49cc-a46b-0f846389a01c.png)

[Error Log](https://astrocolony.com/errorlog/create_componentError.html)
Actually it's what I thought, there is something wrong with encoding Mysql database, where I have utf-8

> Internal Server Error: /create/component/
> Traceback (most recent call last):
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
>     response = get_response(request)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/core/handlers/base.py", line 204, in _get_response
>     response = response.render()
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/response.py", line 105, in render
>     self.content = self.rendered_content
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/response.py", line 83, in rendered_content
>     return template.render(context, self._request)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/backends/django.py", line 61, in render
>     return self.template.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 170, in render
>     return self._render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
>     return self.nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/loader_tags.py", line 150, in render
>     return compiled_parent._render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
>     return self.nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/loader_tags.py", line 62, in render
>     result = block.nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
>     return nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
>     return nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/crispy_forms/templatetags/crispy_forms_tags.py", line 213, in render
>     return template.render(c)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/backends/django.py", line 61, in render
>     return self.template.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 170, in render
>     return self._render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
>     return self.nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/crispy_forms/templatetags/crispy_forms_utils.py", line 21, in render
>     return remove_spaces(self.nodelist.render(context).strip())
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/loader_tags.py", line 195, in render
>     return template.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 172, in render
>     return self._render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
>     return self.nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
>     return nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/loader_tags.py", line 195, in render
>     return template.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 172, in render
>     return self._render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
>     return self.nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/crispy_forms/templatetags/crispy_forms_utils.py", line 21, in render
>     return remove_spaces(self.nodelist.render(context).strip())
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/defaulttags.py", line 211, in render
>     nodelist.append(node.render_annotated(context))
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/loader_tags.py", line 195, in render
>     return template.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 172, in render
>     return self._render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
>     return self.nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
>     return nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
>     return nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
>     return nodelist.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 938, in render
>     bit = node.render_annotated(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
>     return self.render(context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/crispy_forms/templatetags/crispy_forms_field.py", line 155, in render
>     return str(field)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/utils/html.py", line 376, in <lambda>
>     klass.__str__ = lambda self: mark_safe(klass_str(self))
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/forms/boundfield.py", line 34, in __str__
>     return self.as_widget()
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/forms/boundfield.py", line 93, in as_widget
>     return widget.render(
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/forms/widgets.py", line 246, in render
>     context = self.get_context(name, value, attrs)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/forms/widgets.py", line 683, in get_context
>     context = super().get_context(name, value, attrs)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/forms/widgets.py", line 644, in get_context
>     context['widget']['optgroups'] = self.optgroups(name, context['widget']['value'], attrs)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/weblate/utils/forms.py", line 68, in optgroups
>     groups = super().optgroups(name, value, attrs)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/forms/widgets.py", line 592, in optgroups
>     for index, (option_value, option_label) in enumerate(self.choices):
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/forms/models.py", line 1169, in __iter__
>     for obj in queryset:
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/db/models/query.py", line 353, in _iterator
>     yield from self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/db/models/query.py", line 51, in __iter__
>     results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
>     cursor.execute(sql, params)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/db/backends/utils.py", line 98, in execute
>     return super().execute(sql, params)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
>     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
>     return executor(sql, params, many, context)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
>     return self.cursor.execute(sql, params)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 73, in execute
>     return self.cursor.execute(query, args)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/MySQLdb/cursors.py", line 206, in execute
>     res = self._query(query)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/MySQLdb/cursors.py", line 321, in _query
>     self._post_get_result()
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/MySQLdb/cursors.py", line 355, in _post_get_result
>     self._rows = self._fetch_row(0)
>   File "/home/admin/weblate-env/lib/python3.8/site-packages/MySQLdb/cursors.py", line 328, in _fetch_row
>     return self._result.fetch_row(size, self._fetch_type)
**> UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe1 in position 5: unexpected end of data**
> [22/Sep/2021 21:28:31] "GET /create/component/ HTTP/1.0" 500 797626

**Reply by Tefel** [reply]
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe1 in position 5: unexpected end of data
Not sure if it's a server error or a code itself. Please help.

**Comment by Tefel** [comment, marked as answer]
**I was able to resolve the issue by changing the encoding from utf-8 to latin1**
Sorry for the topic. I got a little confused. It was over a decade ago when I looked into python and bash.

**Reply by nijel** [reply]
Asuming you want to translate into non-latin1 languages, it might be better to configure MySQL to use UTF-8 instead, see https://docs.weblate.org/en/latest/admin/install.html#mysql-and-mariadb.

**Reply by Tefel** [reply]
Thanks for suggesting that, right now Chinese (simplified) is also working!
